### ISSUE TITLE
Subscriptions: Replace with SubmitButton new Button block

### ISSUE THREAD

**Issue by scruffian** [Automattic/jetpack#15714]
#### Is your feature request related to a problem? Please describe.

Now we have a [shared button block](https://github.com/Automattic/jetpack/pull/15370), we should update the Subscriptions block to use this instead of the SubmitButton component.

From https://github.com/Automattic/jetpack/issues/14761

**Comment by aaronrobertshaw** [comment]
### TL;DR

* Closing this, after decision to **leave the Subscription block as it is** for now. 
* Not updating to the shared Button **does not prevent removal of the old SubmitButton** component.
* Updating will take considerable work, potentially introduce UX issues and provide little overall benefit.

### Explanation

The Subscriptions block currently uses the RichText component for displaying its button within the editor instead of the SubmitButton. It does this in much the same way as the shared Button actually uses the same RichText component. As a result, this block already offers all the functionality of the shared Button block but also adds the ability to modify the internal padding, font size, border color and border width.

The catch with utilising the shared Button here is that the Subscriptions block currently shares some its attributes across both the text control and button to keep them visually in sync. So in addition to updating the controllable attributes of the shared Button block we'd also need to be adding a mechanism to achieve arbitrary bi-directional synchronising of its attributes. The option of re-writing the Subscriptions block to be entirely block based rather than a wrapper to generate a shortcode, was also explored. This might be a better option in the long run although there are higher priorities requiring attention at the moment and this work can be addressed via a separate issue.

After a few internal discussions, it has been decided to leave the Subscriptions block as it is for now. There doesn't appear to be enough benefit in updating it compared to the work involved. We can still proceed with removing the SubmitButton component however.
